src/main/java/org/opensearch/knn/index/KNNSettings.java

```diff
@@ -262,35 +265,33 @@ public static synchronized KNNSettings state() {
         return INSTANCE;
     }
 
-    public void setSettingsUpdateConsumers() {
-        for (Setting<?> setting : dynamicCacheSettings.values()) {
-            clusterService.getClusterSettings().addSettingsUpdateConsumer(setting, newVal -> {
-                logger.debug("The value of setting [{}] changed to [{}]", setting.getKey(), newVal);
-                latestSettings.put(setting.getKey(), newVal);
+    private void setSettingsUpdateConsumers() {
+        clusterService.getClusterSettings().addSettingsUpdateConsumer(updatedSettings -> {
+            // When any of the dynamic settings are updated, rebuild the cache with the updated values. Use the current
+            // cluster settings values as defaults.
+            boolean isCircuitBreakerEnabled = updatedSettings.getAsBoolean(
+                KNN_MEMORY_CIRCUIT_BREAKER_ENABLED,
+                getSettingValue(KNN_MEMORY_CIRCUIT_BREAKER_ENABLED)
+            );
 
-                // Rebuild the cache with updated limit
-                NativeMemoryCacheManager.getInstance().rebuildCache();
-            });
-        }
+            ByteSizeValue maxCacheWeight = getSettingValue(KNN_MEMORY_CIRCUIT_BREAKER_LIMIT);
```

Comment:
nit: can we change maxCacheWeight type to long, this way it's gonna be consistent with expiryTimeInMinutes setting. now we do have both primitive object and getter of complex. seems maxCacheWeight it's not used anywhere else in this method so ByteSizeValue type is not required

Reply:
Sure, will update.